tests/pruners_tests/test_median.py

```diff
@@ -120,16 +122,18 @@ def test_median_pruner_interval_steps(
     study = optuna.study.create_study()
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
-    n_steps = max(expected_prune_steps)
-    base_index = 0
-    for i in range(base_index, base_index + n_steps):
-        trial.report(base_index, i)
+    last_step = max(expected_prune_steps) + 1
+
+    for i in range(last_step):
+        trial.report(0, i)
     study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
-    for i in range(base_index, base_index + n_steps):
-        if (i - base_index) % report_steps == 0:
+
+    pruned = []
+    for i in range(last_step):
+        if i % report_steps == 0:
             trial.report(2, i)
-        assert pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id)) == (
```

Comment:
This test can pass wrongly even if `expected_prune_steps = [0, 1, 2, 3]` and `pruned = []`.

Reply:
How did you confirm the implementation is correct?

If some test case passes unexpectedly, then it'd be better to check the logic is correct or not.

Reply:
Fortunately, it seems that the implementation is expected to be correct, because those test cases passed after this change. https://github.com/optuna/optuna/pull/2171/files#diff-7945f3f629a92b7cd2958d1ee6b51816c7072f396447485a8db6704a6abbfd7aR110-R113

Meanwhile, I will check the logic is correct or not as I need to understand it for another PR.